docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+
+![ManPage](diagrams/ManPage.png)
+
+**Design Considerations**
+
+**Alternative 1** : Prints the manual page individually as features. 
+Pros: Avoids wordy and long manual page when command is man, ability to have more description
+for specific features. Avoids long methods. 
+Cons: Extra step of typing another command to understand a feature well.
+
+**Alternative 2**: Combining all feature descriptions in a single manual page. 
+Pros: Avoid extra step of typing another command to understand a feature well.
+Cons: Wordy and long manual page when command is man, user has to scroll through
+a long manual page to find what they need. Long method.
+
+### 3.2. Parser
+
+The `Parser` is a class that handles the commands keyed in by the user.  
+It has multiple checks to ensure that the user keys in valid commands. 
+
+<!---elaborate more on the class, still unsure of the format...--->
+<!---insert other commands here, still unsure of the format...(should it all be under Parser class or separate?)--->
+
+### 3.3. Add Modules
+
+`add` command:
+
+The `add` command is used by the user to add a module or multiple modules at once.
+
+The sequence by which the Parser class handles the `add` command is as follows:
+1) The `Parser` class extracts the necessary fields from the user input, and calls upon the `addModule`
+   method in the `ModuleList` class in a loop.
+2) The method then checks the type of module that is being added, then calls the relevant
+   constructor (`Core`, `UE`, `GE`, `Internship`).
+3) The `addModule` method returns the module that is added to the `Parser` class.
+4) The `getModuleListSize` method is called from `ModuleList` class.
+5) The `printAddedModule` method is called from the `Print` class to display the result to the user.
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![addInputCommand](diagrams/AddInputCommand.png)
+
+<small><i>Figure ???</i></small>
+
+### 3.4. Find Modules
+
+The find modules features is facilitated by the `Parser`, `ModuleList` and `Module` Classes. `Parser` helps to check
+for user input to determine is user doing a search via a module name or module type, which will then invoke the 
+`findModuleByName` and `findModuleByType` methods respectively. If any modules are found, the ChatBot will print out
+all the modules, else the ChatBot will return a message to the user, saying that no modules were found with the search
+term they used.
+
+Given below is an example usage scenario and how the findModuleByName mechanism behaves at each step.
+
+Step 1: The user launches the application for the first time. The ChatBot will be initialised with an empty ArrayList
+of Modules.
+
+Step 2: The user executes `add /CS2113T CS2101 /4 /CORE /2 /2` to add two modules into the ChatBot via the addModules 
+feature
+
+The following object diagram shows the current state of the ChatBot:
+![FindModule](diagrams/FindTwoModules.png)
```

Comment:
1. It may be clearer if the object that starts all of it is named `Modganiser` instead of `Duke`.
2. Since a Module object can contain attributes like: modular credits, year and semester, you may want to include them in the attribute section of the Module objects, since they are mentioned in step 2